src/com/elikill58/negativity/spigot/packets/PacketType.java

```diff
@@ -0,0 +1,296 @@
+			for(Client client : Client.values())
+				if(client.getFullName().equalsIgnoreCase(packetName))
+					return client;
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow client packet " + packetName);
+			return Client.UNSET;
+		} else if(packetName.startsWith(SERVER_PREFIX)) {
+			for(Server srv : Server.values())
+				if(srv.getFullName().equalsIgnoreCase(packetName))
+					return srv;
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow server packet " + packetName);
+			return Server.UNSET;
+		} else if(packetName.startsWith(LOGIN_PREFIX)) {
+			for(Login login : Login.values())
+				if(login.getFullName().equalsIgnoreCase(packetName))
+					return login;
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow login packet " + packetName);
+			return Login.UNSET;
+		} else if(packetName.startsWith(STATUS_PREFIX)) {
+			for(Status status : Status.values())
+				if(status.getFullName().equalsIgnoreCase(packetName))
+					return status;
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow status packet " + packetName);
+			return Status.UNSET;
+		} else {
+			if(LOG_UNKNOW_PACKET)
+				SpigotNegativity.getInstance().getLogger().info("[Packet] Unknow packet " + packetName);
+			return null;
+		}
+	}
+	
+	public static enum Client implements PacketType {
+		ABILITIES("Abilities"),
+		ADVANCEMENTS("Advancements"),
+		ARM_ANIMATION("ArmAnimation"),
+		AUTO_RECIPE("AutoRecipe"),
+		BEACON("Beacon"),
+		BEDIT("BEdit"),
+		BOAT_MOVE("BoatMove"),
+		BLOCK_DIG("BlockDig"),
+		BLOCK_PLACE("BlockPlace"),
+		CHAT("Chat"),
+		CLIENT_COMMAND("ClientCommand"),
+		CLOSE_WINDOW("CloseWindow"),
+		CUSTOM_PAYLOAD("CustomPayload"),
+		DIFFICULTY_CHANGE("DifficultyChange"),
+		DIFFICULTY_LOCK("DifficultyLock"),
+		ENCHANT_ITEM("EnchantItem"),
+		ENTITY_ACTION("EntityAction"),
+		ENTITY_NBT_QUERY("EntityNBTQuery"),
+		FLYING("Flying"),
+		HELD_ITEM_SLOT("HeldItemSlot"),
+		ITEM_NAME("ItemName"),
+		KEEP_ALIVE("KeepAlive"),
+		LOOK("Look"),
+		PICK_ITEM("PickItem"),
+		POSITION("Position"),
+		POSITION_LOOK("PositionLook"),
+		RECIPE_DISPLAYED("RecipeDisplayed"),
+		RESOURCE_PACK_STATUS("ResourcePackStatus"),
+		SET_COMMAND_BLOCK("SetCommandBlock"),
+		SET_COMMAND_MINECART("SetCommandMinecart"),
+		SET_CREATIVE_SLOT("SetCreativeSlot"),
+		SET_JIGSAW("SetJigsaw"),
+		SETTINGS("Settings"),
+		SPECTATE("Spectate"),
+		STEER_VEHICLE("SteerVehicle"),
+		STRUCT("Struct"),
+		TAB_COMPLETE("TabComplete"),
+		TELEPORT_ACCEPT("TeleportAccept"),
+		TILE_NBT_QUERY("TileNBTQuery"),
+		TR_SEL("TrSel"),
+		TRANSACTION("Transaction"),
+		UPDATE_SIGN("UpdateSign"),
+		USE_ENTITY("UseEntity"),
+		USE_ITEM("UseItem"),
+		VEHICLE_MOVE("VehicleMove"),
+		WINDOW_CLICK("WindowClick"),
+		UNSET("Unset");
+		
+		private final String packetName, fullName;
+		
+		private Client(String packetName) {
```

Comment:
```suggestion
		Client(String packetName) {
```
Pareil pour les autres enums